### PR TITLE
[CAS-905] Refac - Removing firebase from chat domain

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
[CAS-905](https://stream-io.atlassian.net/browse/CAS-905)

### 🎯 Goal

Remove Firebase dependency in offline module. 

### 🛠 Implementation details

Create a BroadcastReceiver to sync all messages once a broadcast is sent after new push notifications arrive. The old service is no longer used and instead we are using a WorkManager to perform the sync of the messages. 

### 🧪 Testing

Send new push notification and see if the messages are synced. It is also needed to test the app in the Doze mode. 

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog is updated with client-facing changes. Todo
- ~[ ] New code is covered by unit tests~
- ~[ ] Comparison screenshots added for visual changes~
- [ ] Affected documentation updated (CMS, cookbooks, tutorial). Todo
- [] Reviewers added

### 🎉 GIF

giphy

### CODE REVIEW

File: stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/Push.java [hunk 1]
```diff
@@ -110,26 +109,26 @@ public Intent getNewMessageIntent(
...
-            if (pushDataSyncHandler.isStreamMessage(message)) {
-                // handle RemoteMessage sent from Stream backend
-                pushDataSyncHandler.onMessageReceived(message);
-            } else {
-                // handle RemoteMessage from other source
-            }
-            stopSelf();
-        }
-    }
+//    class CustomFirebaseMessagingService extends FirebaseMessagingService {
```
Comment (leandroBorgesFerreira) [inline stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/Push.java:112]: I hava to update this documentation

File: stream-chat-android-offline/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -1,15 +1,9 @@
...
-        <service
-            android:name="io.getstream.chat.android.livedata.service.sync.OfflineSyncFirebaseMessagingService"
-            android:exported="false"
-            >
-            <intent-filter android:priority="-1" >
-                <action android:name="com.google.firebase.MESSAGING_EVENT" />
-            </intent-filter>
-        </service>
+        <receiver android:name=".service.sync.OfflineSyncFirebaseMessagingReceiver">
+<!--            <intent-filter Todo: declare it!-->
```
Comment (leandroBorgesFerreira) [inline stream-chat-android-offline/src/main/AndroidManifest.xml:6]: Once a have the intent, update here

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/service/sync/PushMessageSyncHandler.kt [hunk 3]
```diff
@@ -1,176 +0,0 @@
...
-        client: ChatClient,
-        message: RemoteMessage,
-    ) {
-        val result = domain.replayEventsForActiveChannels(cid).execute()
-        if (result.isSuccess) {
-            logger.logD("Sync success.")
-        } else {
-            logger.logD("Sync failed.")
-        }
-        client.onMessageReceived(message)
```
Comment (leandroBorgesFerreira) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/service/sync/PushMessageSyncHandler.kt:125]: This needs to be moved to LLC

**Review by JcMinarro — DISMISSED** [review]
Just blocked for now

**Review by adasiewiczr — APPROVED** [review]

**Pull request merged**